### DISCUSSION TITLE
WhatsApp processing in Cruiser

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#467] [Q&A] [answered]
As already may be known, I put all the information for a tour in a Dropbox directory that is shared with all participants. Now there are people who refuse to use Dropbox and want to receive a route via WhatsApp. There are people ...

When I send the short link via WhatsApp, the recipient can open the route in the Kurviger app by simply tapping on it - Cruiser is even not displayed in the list of apps to be used:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/e7e2c48f-e5e9-44d6-a4ba-0d0d830d14f6"/>  <img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/0a721e58-7f43-4c7d-a2f5-31e799a86423"/>  <img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/b0987756-a1fa-4e61-8978-1d7d5d60bd13"/>
 
I can also send the *.kurviger file as a "Document". It can be opened in Kurviger, in Cruiser you get an error message:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/c3659e13-d2bf-4993-b709-01ecca69c64d"/>  <img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/9d3370a4-0ab7-4739-a7a0-13e3ef6d13c0"/>   <img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/42786906-19ec-448e-a6c1-c397cf4b7890"/>

As a workaround, however, I can load the file from the depths of Android via /Routing/Import: I find it in /WhatsApp/Media/WhatsAppDocuments/Sent. 

Is it possible to implement the admittedly simpler solution of the Kurviger app in Cruiser?

**Comment by devemux86** [comment]
@WolfgangZ42 thanks for the suggestion.

> When I send the short link

Short links do not contain offline routes, they contain full links that require online route calculation.
So they cannot be used by other apps.

> in Cruiser you get an error message

We need to check this. :slightly_smiling_face:

**Comment by devemux86** [comment]
@WolfgangZ42 thanks for the detailed report! :+1: 

> in Cruiser you get an error message

This should be fixed in the latest:
- https://github.com/devemux86/cruiser/discussions/451

**Comment by devemux86** [comment]
For route sharing we need a practical solution, even better without internet:
- https://github.com/devemux86/cruiser/discussions/330

**Comment by MotoUKRider** [comment]
The cloud/sharing feature sounds like the sort of thing a few of the other apps might offer (not to mention a revenue stream for the author/hosting!)... A way for *heavy* users to push their stuff to the cloud, but allow anyone (ie: their friends/tour-clients) to download it "in app" without special privileges.  A future consideration, possibly :smile: 

I navigate my way through Dropbox (via Firefox) on the phone when updating the Cruiser Betas.  There are some "close/disregard" actions to perform (because websites try to make your life easier by being annoying)... No account/sign-in, I then launch it from "CX File Explorer" (a nice file app with network features so I can exchange files with the desktop PC).

For offline (car-park/cafe) sharing, I'd say that [my suggestion of a simple Bluetooth routine](https://github.com/devemux86/cruiser/discussions/330#discussioncomment-8591771) might be the most sensible.

**Comment by devemux86** [comment, marked as answer]
Fixed in [Cruiser 3.0.41](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
